packages/ui/src/multiplying-architecture/KaotoEditorFactory.ts

```diff
@@ -6,12 +6,25 @@ import {
   KogitoEditorEnvelopeContextType,
 } from '@kie-tools-core/editor/dist/api';
 import { KaotoEditorChannelApi } from './KaotoEditorChannelApi';
+import { CatalogSchemaLoader, isDefined } from '../utils';
+import { LocalStorageSettingsAdapter } from '../models/settings/localstorage-settings-adapter';
 
 export class KaotoEditorFactory implements EditorFactory<Editor, KaotoEditorChannelApi> {
-  public createEditor(
+  public async createEditor(
     envelopeContext: KogitoEditorEnvelopeContextType<KaotoEditorChannelApi>,
     initArgs: EditorInitArgs,
   ): Promise<Editor> {
-    return Promise.resolve(new KaotoEditorApp(envelopeContext, initArgs));
+    let catalogUrl;
+    const catalogUrlFromChannelApi = await envelopeContext.channelApi.requests.getCatalogURL();
+    if (isDefined(catalogUrlFromChannelApi)) {
+      catalogUrl = catalogUrlFromChannelApi;
+    } else {
+      catalogUrl = `${initArgs.resourcesPathPrefix}${CatalogSchemaLoader.DEFAULT_CATALOG_PATH.replace('.', '')}`;
+    }
+    const settingsAdapter = new LocalStorageSettingsAdapter();
+    const settingsModel = settingsAdapter.getSettings();
+    settingsModel.catalogUrl = catalogUrl;
+    settingsAdapter.saveSettings(settingsModel);
```

Comment:
At this moment, this is not doing anything yet, because the SettingsAdapter is created somewhere else and this one is not used.

Reply:
the LocalStorageSettingsAdapter is storing data in the localStorage https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage so I was expecting it is shared across all classes from the same window

Reply:
anyway, do not need it as the Settings are no more used in the runtimeProvider